optaplanner-examples/src/main/java/org/optaplanner/examples/examination/solver/score/ExaminationConstraintProvider.java

```diff
@@ -207,15 +223,19 @@ protected Constraint frontLoad(ConstraintFactory constraintFactory) {
     }
 
     protected Constraint periodPenalty(ConstraintFactory constraintFactory) {
-        return constraintFactory.from(Exam.class)
-                .filter(exam -> exam.getPeriod().getPenalty() != 0)
-                .penalizeConfigurable("periodPenalty", exam -> exam.getPeriod().getPenalty());
```

Comment:
I prefer the original rule.
It reads better and it's probably faster too (no joins).

Reply:
This is actually a performance improvement - there is only a small number of `Period`s, and many of them will be ignored due to the penalty being zero. Therefore, the subsequent `join` will result in *much less* tuples that have to be traversed, than if I start will all the `Exam`s.

I'm keeping this as is.